internal/cmd/agent/deployer/monitor/updatestatus_test.go

```diff
@@ -0,0 +1,208 @@
+				assert.True(t, status.ModifiedStatus[0].Create)
+				assert.Emptyf(t, status.NonReadyStatus, "expected non-ready status to be empty")
+			},
+		},
+		{
+			name: "modified",
+			args: args{
+				resources: []fleet.BundleDeploymentResource{
+					{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testcm",
+					},
+				},
+				modified: []fleet.ModifiedStatus{
+					{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testcm",
+						Patch:      `{"data": {"foo": "bar"}`,
+					},
+				},
+			},
+			assert: func(t *testing.T, status fleet.BundleDeploymentStatus) {
+				assert.Equal(t, status.ResourceCounts, fleet.ResourceCounts{DesiredReady: 1, Modified: 1})
+				assert.Truef(t, status.Ready, "unexpected ready status")
+				assert.Falsef(t, status.NonModified, "unexpected non-modified status")
+				assert.Lenf(t, status.Resources, 1, "unexpected resources length")
+				assert.Len(t, status.ModifiedStatus, 1, "incorrect modified status length")
+				assert.NotEmpty(t, status.ModifiedStatus[0].Patch)
+				assert.Emptyf(t, status.NonReadyStatus, "expected non-ready status to be empty")
+			},
+		},
+		{
+			name: "missing and non-ready",
+			args: args{
+				resources: []fleet.BundleDeploymentResource{
+					{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testcm",
+					},
+					{
+						Kind:       "Pod",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testpod",
+					},
+				},
+				modified: []fleet.ModifiedStatus{
+					{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testcm",
+						Patch:      `{"data": {"foo": "bar"}`,
+					},
+				},
+				nonReady: []fleet.NonReadyStatus{
+					{
+						Kind:       "Pod",
+						APIVersion: "v1",
+						Namespace:  "testns",
+						Name:       "testpod",
+						Summary: fleetv1.Summary{
+							State:   "Evicted",
+							Error:   true,
+							Message: []string{"no space left on device"},
+						},
+					},
+				},
+			},
+			assert: func(t *testing.T, status fleet.BundleDeploymentStatus) {
+				assert.Equal(t, status.ResourceCounts, fleet.ResourceCounts{DesiredReady: 2, Modified: 1, NotReady: 1})
+				assert.Falsef(t, status.Ready, "unexpected ready status")
+				assert.Falsef(t, status.NonModified, "unexpected non-modified status")
+				assert.Lenf(t, status.Resources, 2, "unexpected resources length")
+				assert.Len(t, status.NonReadyStatus, 1, "incorrect non-ready status length")
+				assert.Equal(t, status.NonReadyStatus[0].Summary, fleetv1.Summary{
+					State:   "Evicted",
+					Error:   true,
+					Message: []string{"no space left on device"},
+				})
+				assert.Len(t, status.ModifiedStatus, 1, "incorrect modified status length")
+				assert.NotEmpty(t, status.ModifiedStatus[0].Patch)
```

Comment:
nit: in the same way that we check the summary of the first `status.NonReadyStatus` element here, we could check the contents of `status.ModifiedStatus[0].Patch`.

Reply:
I actually didn't like much to be so specific, since it's probably outside of the scope for the test. I may simplify the case for Summary instead. WDYT?

Reply:
Less specific also works :)